src/System.IO.FileSystem/src/System/IO/FindPredicate.cs

```diff
@@ -7,5 +7,5 @@ namespace System.IO
     /// <summary>
     /// Interface for filtering out find results.
     /// </summary>
-    internal delegate bool FindPredicate<TState>(ref RawFindData findData, TState state);
+    internal delegate bool FindPredicate<TState>(ref RawFindData<TState> findData);
```

Comment:
If both the delegate types have the same signature now, how about just defining a delegate:
```C#
internal delegate bool FindFunc<TState>(ref RawFindData<TState> findData);
```
and using the same delegate for both?